storage/ops.go

```diff
@@ -1793,52 +1796,42 @@ func validatePartitions(rootSize uint64, medias []*BlockDevice, legacyBios bool,
 					log.Warning("validatePartitions: Skipping swap size check due to skipSize")
 				} else {
 					if ch.Size < swapSize {
-						size, _ := HumanReadableSizeWithPrecision(swapSize, 1)
-						warning := utils.Locale.Get("%s must be %s", swapLabel,
-							fmt.Sprintf(">= %s", size))
-						results = append(results, warning)
-						log.Warning("validatePartitions: %s %+v", warning, ch)
+						results = append(results, logPartitionSizeWarning(ch, swapSize, swapLabel))
 					}
 				}
 			}
 		}
 	}
 
 	if !rootFound || rootBlockDevice == nil {
-		warning := utils.Locale.Get("Missing %s partition", rootLabel)
-		results = append(results, warning)
-		log.Warning("validatePartitions: %s", warning)
+		results = append(results, logMissingPartition(rootLabel))
 	}
 
 	if !bootFound {
-		if legacyBios && skipAll {
-			if rootBlockDevice != nil {
-				if !(rootBlockDevice.FsType == "ext2" || rootBlockDevice.FsType == "ext3" ||
-					rootBlockDevice.FsType == "ext4") {
-					// xfs currently not supported due to partition table of MBR requirement
-					warning := utils.Locale.Get("%s must be %s", rootLabel, "ext[234]")
-					results = append(results, warning)
-					log.Warning("validatePartitions: legacyMode, invalid fsType: %s %s %+v",
-						rootBlockDevice.FsType, warning, rootBlockDevice)
-				}
-				if rootBlockDevice.Type == BlockDeviceTypeCrypt {
-					warning := utils.Locale.Get("Encryption of %s is not supported", rootLabel)
-					results = append(results, warning)
-					log.Warning("validatePartitions: legacyMode without /boot, %s %+v",
-						warning, rootBlockDevice)
+		if skipAll {
```

Comment:
why did we break it down one level.... it could have remained the same as before:   if legacyBios && SkipAll

Reply:
No, that logic doesn't work (that was my first approach which didn't cover all the cases)
We need:

- No Boot Found, Skip Validation Requested, **and** Legacy BIOS mode (We still need to check root partition)
- No Boot Found, Skip Validation Requested, **and** NOT Legacy BIOS mode (Error: Must have /boot if not Legacy mode)
- No Boot Found, NOT Skip Validation Requested (Error: Must have /boot if not Legacy mode)

Reply:
Is it possible to move sections of code to another helper functions, This function is getting increasily tough to read.

Reply:
@karthikprabhuvinod  Hence #530

Reply:
@mdhorn Sure will take care of it during #530